tests/ui/check-cfg/compact-names.rs

```diff
@@ -13,4 +13,8 @@ pub fn expected() {}
 //~^ WARNING unexpected `cfg` condition name
 pub fn unexpected() {}
 
+#[cfg(target(os = "windows", architecture = "arm"))]
+//~^ WARNING unexpected `cfg` condition name
```

Comment:
The problem was that I implemented evaluation of `all()` and `any()` as short-circuiting.
This is not correct because during evaluation, sometimes lints are emitted (see https://github.com/JonathanBrouwer/rust/blob/7a7c74ad89df9f87824fa17fbbe0448d9ab6f7cc/compiler/rustc_attr_parsing/src/attributes/cfg.rs#L233). We should emit these lints, even after an earlier condition already failed.

The only test that tests this apparently is the `unexpected` test above this comment, which happens to only test this for non-linux targets. I added a second test `unexpected2` to also test this on linux targets.